src/main/java/gov/cms/madie/cqllibraryservice/config/DeleteTestLibrariesChangeUnit.java

```diff
@@ -0,0 +1,168 @@
+import gov.cms.madie.cqllibraryservice.repositories.LibrarySetActionLogRepository;
+import gov.cms.madie.cqllibraryservice.repositories.LibrarySetRepository;
+import gov.cms.madie.models.common.ActionLog;
+import gov.cms.madie.models.common.LibraryActionLog;
+import gov.cms.madie.models.common.LibrarySetActionLog;
+import gov.cms.madie.models.library.LibrarySet;
+import gov.cms.madie.models.library.CqlLibrary;
+import io.mongock.api.annotations.ChangeUnit;
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@ChangeUnit(id = "delete_test_libraries", order = "1", author = "madie_dev")
+public class DeleteTestLibrariesChangeUnit {
+  private final List<String> users =
+      Arrays.asList("cvasile", "bwelch", "pvasireddy", "colin.sullivan", "adongare");
+  private final Set<String> userSet = new HashSet<>(users);
+
+  private List<LibraryActionLog> filteredActionLogs = new ArrayList<>();
+  private List<LibrarySetActionLog> filteredLibrarySetActionLogs = new ArrayList<>();
+
+  private List<LibrarySet> filteredLibrarySets = new ArrayList<>();
+  private List<CqlLibrary> filteredLibraries = new ArrayList<>();
+
+  @Execution
+  public void deleteTestLibraries(
+      CqlLibraryRepository cqlLibraryRepository,
+      LibrarySetRepository librarySetRepository,
+      ActionLogRepositoryImpl actionLogRepository,
+      LibrarySetActionLogRepository librarySetActionLogRepository) {
+
+    // 1. delete ActionLogs and CqlLibraryLogs
+    deleteActionLogs(actionLogRepository);
+    deleteLibrarySetActionLogs(librarySetActionLogRepository);
+
+    // 2. delete CqlLibrary and LibrarySet
+    List<LibrarySet> librarySets = librarySetRepository.findAll();
+    filteredLibrarySets =
+        librarySets.stream().filter(ms -> userSet.contains(ms.getOwner())).toList();
+    if (CollectionUtils.isNotEmpty(filteredLibrarySets)) {
+      List<String> filteredLibrarySetIds =
+          filteredLibrarySets.stream().map(LibrarySet::getLibrarySetId).toList();
+      filteredLibraries = cqlLibraryRepository.findByLibrarySetIdIn(filteredLibrarySetIds);
+
+      deleteLibraries(cqlLibraryRepository, filteredLibraries);
+      deleteLibrarySets(librarySetRepository, filteredLibrarySets);
+    }
+  }
+
+  void deleteActionLogs(ActionLogRepository actionLogRepository) {
+    checkActionLogs(actionLogRepository);
+    actionLogRepository.removeActionsByUsers(users, ActionLog.class);
+  }
+
+  void deleteLibrarySetActionLogs(LibrarySetActionLogRepository librarySetActionLogRepository) {
+    checkLibrarySetActionLogs(librarySetActionLogRepository);
+    librarySetActionLogRepository.removeActionsByUsers(users, LibrarySetActionLog.class);
+  }
+
+  // for logging purpose
+  void checkActionLogs(ActionLogRepository actionLogRepository) {
+    List<LibraryActionLog> actionLogs = actionLogRepository.findAllActionLogs();
+    log.info("ActionLog total = " + actionLogs.size());
```

Comment:
Minor edit throughout: Prefer parameterized log messages over concatenation.
```suggestion
    log.info("ActionLog total = {}", actionLogs.size());
```
It's not critical, especially since this is a one off script. Parameterized logging reduces the number of Strings being generated since the concatenation will occur even if the log level is set higher than the log statement.